### PR TITLE
feat: splitbutton menu enhancement

### PR DESCRIPTION
**Pull request by elizeumatheus**
# SplitButton - Menu Enchancement

## Menu Fit Button Container
Now we have a property to set the width of menu container like the button container.
image

## Menu Open Style
Now we can switch between `hover` or `click` action to open the menu.

### Menu Open on Hover
Kapture 2020-06-30 at 16 10 18

### Menu Open on Click
Kapture 2020-06-30 at 16 16 14

### CODE REVIEW

File: packages/react-ds/src/components/SplitButton/SplitButton.tsx [hunk 1]
```diff
@@ -16,17 +16,26 @@ const SplitButton: React.FC<ISplitButtonProps> = ({
...
   ...props
 }: ISplitButtonProps) => {
   const [isOpen, setIsOpen] = React.useState(false)
 
+  const clickType = openType === 'click' && isOpen ? styles.active : ''
+
   return (
-    <nav className={styles.splitButton} {...props}>
+    <nav
+      className={`${styles.splitButton}
```
Comment (carloshrabelo) [inline packages/react-ds/src/components/SplitButton/SplitButton.tsx:29]: Can use classnames
```
className={classNames(styles.splitButton, styles[openType], {
    [styles.active]: openType === "click" && isOpen,
  })}

File: packages/react-ds/src/components/SplitButton/SplitButton.tsx [hunk 2]
```diff
@@ -36,7 +45,7 @@ const SplitButton: React.FC<ISplitButtonProps> = ({
                     ${styles[size]}
                     ${styles[color]}
                     ${styles[direction]}
-                    ${isOpen ? styles.active : ''}`}
+                    ${menuFitButtonContainer ? styles.fitbutton : ''}`}
```
Comment (carloshrabelo) [inline packages/react-ds/src/components/SplitButton/SplitButton.tsx:48]: use classnames
```
  className={classNames(
    styles.dropdown,
    styles[size],
    styles[color],
    styles[direction],
    { [styles.fitbutton]: menuFitButtonContainer }
  )}

File: packages/core/models/split-button.model.ts [hunk 3]
```diff
@@ -7,4 +7,8 @@ export interface ISplitButton {
   text: string
   /** ltr | rtl */
   direction?: 'ltr' | 'rtl'
+  /** boolean */
+  menuFitButtonContainer?: boolean
```
Comment (felipefialho) [inline packages/core/models/split-button.model.ts:11]: I thing that is a big big name for a prop, hahaha 🤔 

Maybe just... `isFitMenu`?

Comment (elizeumatheus) [reply packages/core/models/split-button.model.ts]: Made on 5477daf

File: packages/react-ds/src/components/SplitButton/SplitButton.tsx (hunk 2 above)
Comment (elizeumatheus) [reply packages/react-ds/src/components/SplitButton/SplitButton.tsx]: Made on 962ae89

File: packages/react-ds/src/components/SplitButton/SplitButton.tsx (hunk 1 above)
Comment (elizeumatheus) [reply packages/react-ds/src/components/SplitButton/SplitButton.tsx]: Made on 962ae89

File: packages/react-ds/src/components/SplitButton/SplitButton.stories.mdx [hunk 4]
```diff
@@ -58,6 +64,88 @@ import SplitButton from './SplitButton'
   </Story>
 </Preview>
 
+## Menu
+
+### Menu Fit Button Container
+
+<Preview>
+  <Story name="Menu Fit Button Container" height="200px">
```
Comment (felipefialho) [inline packages/react-ds/src/components/SplitButton/SplitButton.stories.mdx:72]: Maybe we don't need a story for it, just a `knob`, do you agree? 🤔

File: packages/styles/components/SplitButton.module.scss [hunk 5]
```diff
@@ -45,9 +46,13 @@
   will-change: visibility;
   z-index: var(--zindex-10);
 
+  .hover > & {
+    transition-delay: 0.2s;
+  }
```
Comment (felipefialho) [inline packages/styles/components/SplitButton.module.scss:51]: 😁

File: packages/styles/components/SplitButton.module.scss [hunk 6]
```diff
@@ -91,7 +96,16 @@
...
+  &.fitbutton {
+    width: 100%;
+
+    > * {
+      white-space: normal;
+    }
+  }
+
+  .hover:hover > &,
+  .click.active > & {
```
Comment (felipefialho) [inline packages/styles/components/SplitButton.module.scss:108]: 😍

File: packages/vue-ds/src/components/SplitButton/SplitButton.stories.mdx [hunk 7]
```diff
@@ -63,7 +69,96 @@ SplitButton allow user to have many options to choice to take an action.
...
+    }}
+  </Story>
+</Preview>
+
+## Menu
+
+### Menu Fit Button Container
+
+<Preview>
+  <Story name="Menu Fit Button Container" height="225px">
```
Comment (felipefialho) [inline packages/vue-ds/src/components/SplitButton/SplitButton.stories.mdx:82]: Same here..

File: packages/react-ds/src/components/SplitButton/SplitButton.stories.mdx (hunk 4 above)
Comment (elizeumatheus) [reply packages/react-ds/src/components/SplitButton/SplitButton.stories.mdx:72]: Of course my friend

File: packages/vue-ds/src/components/SplitButton/SplitButton.stories.mdx (hunk 7 above)
Comment (elizeumatheus) [reply packages/vue-ds/src/components/SplitButton/SplitButton.stories.mdx]: Made on 8e55222

File: packages/react-ds/src/components/SplitButton/SplitButton.stories.mdx (hunk 4 above)
Comment (elizeumatheus) [reply packages/react-ds/src/components/SplitButton/SplitButton.stories.mdx]: Made on 8e55222

**Pull request merged**